open-api/src/main/resources/dipa-hub.yaml

```diff
@@ -30,10 +30,10 @@ paths:
     $ref: "paths/projectData.yml"
   /timelines/{timelineId}/project/update:
     $ref: "paths/updateProjectData.yml"
-  /timelines/{timelineId}/templates:
-    $ref: "paths/templates.yml"
-  /timelines/{timelineId}/templates/{templateId}:
-    $ref: "paths/updateTemplate.yml"
+  /timelines/{timelineId}/timelineTemplates:
+    $ref: "paths/timelineTemplates.yml"
+  /timelines/{timelineId}/timelineTemplates/{timelineTemplateId}:
+    $ref: "paths/updateTimelineTemplate.yml"

```

Comment:
Aktuell ist die API hier an div. Stellen nicht REST konform, da HTTP-Verbs bspw. nicht korrekt eingesetzt werden.

Beispiel:

```yaml
  /timelines/{timelineId}/increments:
    $ref: "paths/increments.yml"
  /timelines/{timelineId}/increments/add:
    $ref: "paths/addIncrement.yml"
  /timelines/{timelineId}/increments/delete:
    $ref: "paths/deleteIncrement.yml"
```
Sowohl addIncrements, als auch deleteIncrement haben "PATCH" als Verb. Eigentlich nach Rest-Logik wäre das alles ein "path" mit unterschiedlichen Verbs.

Also 
```yaml
  /timelines/{timelineId}/increments:
    $ref: "paths/increments.yml"
```
mit 
```yaml
get:
  ....
put:
  ....
delete:
  ....
```
Gleiches gilt für die `/update` Paths. Das sind eigentlich `PATCH` auf der entsprechenden Resource. Umgekehrt wäre die timeline-ops eher ein `/command`  mit `PUT` weil man ein komplexes Commando ablegt und nicht einfach Werte der "Timeline" ändert. 

@Bock4Soft an der Stelle bin ich in den Timeline-Ops darüber gestolpert, das der OperationType einfach freie String sind, warum nicht als Enum, dann wäre die Kommunikation irgendwie sicher gestellt:

```java
        switch (inlineObject.getOperation()) {
            case "moveTimeline": timelineService.moveTimelineByDays(timelineId, inlineObject.getDays());
                break;
            case "moveTimelineStart": timelineService.moveTimelineStartByDays(timelineId, inlineObject.getDays());
                break;
            case "moveTimelineEnd": timelineService.moveTimelineEndByDays(timelineId, inlineObject.getDays());
                break;
            case "moveMilestone": timelineService.moveMileStoneByDays(timelineId, inlineObject.getDays(), inlineObject.getMovedMilestoneId());
                break;
            default:
                return ResponseEntity.notFound().build();
        }
```

-